make_prg/subcommands/update.py

```diff
@@ -0,0 +1,187 @@
+                prg_builder_tree_node = prg_builder_for_locus.get_node_given_interval(
+                    update_data.ml_path_node_key
+                )
+                prg_builder_tree_node.add_data_to_batch_update(
+                    update_data
+                )
+                leaves_to_update.add(prg_builder_tree_node)
+                nb_of_variants_sucessfully_updated += 1
+            except LeafNotFoundException as exc:
+                logger.warning(f"Failed finding leaf: {exc}")
+                nb_of_variants_with_failed_update += 1
+
+        # update the modified leaves
+        # Note: the sorted() is needed for determinism so that we can compare indexes and test outputs
+        for leaf in sorted(leaves_to_update, key=lambda node: node.id):
+            leaf.batch_update()
+        logger.debug(
+            f"Updated {locus_name}: {nb_of_variants_sucessfully_updated} denovo sequences added!"
+        )
+    else:
+        logger.debug(f"{locus_name} has no new variants, no update needed")
+
+    # regenerate PRG
+    temp_dir = io_utils.get_temp_dir_for_multiprocess(options.temp_dir)
+    locus_prefix = temp_dir / locus_name
+
+    logger.info(f"Writing output files of locus {locus_name}")
+    prg = prg_builder_for_locus.build_prg()
+    prg_builder_for_locus.write_prg(str(locus_prefix), prg)
+    gfa.GFA_Output.write_gfa(str(locus_prefix), prg)
+    prg_builder_for_locus.serialize(f"{locus_prefix}.pickle")
+    with open(f"{locus_prefix}.stats", "w") as stats_filehandler:
+        print(
+            f"{locus_name} {nb_of_variants_sucessfully_updated} {nb_of_variants_with_failed_update}",
+            file=stats_filehandler,
+        )
+    if options.output_graphs:
+        prg_builder_for_locus.output_debug_graphs(Path(options.output_prefix + "_debug_graphs"))
+
+
+def run(cl_options):
+    global options
+    options = cl_options
+    global update_shared_data
+
+    if io_utils.output_files_already_exist(options.output_prefix):
+        raise RuntimeError("One or more output files already exists, aborting run...")
+
+    # read input data
+    logger.info("Checking Multiple Sequence Aligner...")
+    output_dir = Path(options.output_prefix).parent
+    output_dir.mkdir(parents=True, exist_ok=True)
+    temp_dir = io_utils.create_temp_dir(output_dir)
+    options.temp_dir = temp_dir
+    msa_temp_path = temp_dir / "msa_temp"
+    mafft_aligner = MAFFT(executable=options.mafft, tmpdir=msa_temp_path)
+
+    prg_builder_zip_db = None
+    try:
+        logger.info("Reading update data structures...")
+        prg_builder_zip_db = PrgBuilderZipDatabase(options.update_DS)
+        prg_builder_zip_db.load()
+        logger.info(f"Reading {options.denovo_paths}...")
+        denovo_variants_db = DenovoVariantsDB(options.denovo_paths)
+        update_shared_data = UpdateSharedData(denovo_variants_db, mafft_aligner)
+
+        output_dir = Path(options.output_prefix).parent
+        os.makedirs(output_dir, exist_ok=True)
```

Comment:
Why not use the pathlib `mkdir` function?
```suggestion
        output_dir.mkdir(exist_ok=True)
```

Reply:
Agreed